### ISSUE TITLE
About batch_size

### ISSUE THREAD

**Issue by jizeyu** [ZhengyuZhao/Adaptive-Discriminative-Region-Discovery#9]
When I run the  file "dis_map_generation", I adjusted the batch_size to 64, but there will be an error "size mismatch", Can batch_size only be set to 256? and why?

**Comment by ZhengyuZhao** [comment]
> When I run the file "dis_map_generation", I adjusted the batch_size to 64, but there will be an error "size mismatch", Can batch_size only be set to 256? and why?

Could you please provide more details about the error, e.g., which line and if there is an error for each batch of the images or just for some specific batch.

**Comment by jizeyu** [comment]
When I adjusted the batch_size to 64,
![捕获](https://user-images.githubusercontent.com/30789945/90303604-e0475300-dee1-11ea-8a02-ae6debd810d6.JPG)
It looks like "logits = linear_layer(nn.AvgPool2d(kernel_size=14)(conv_maps).view(-1, 512))" have some problem, and I don't understand what m1 and m2 mean here？

**Comment by ZhengyuZhao** [comment]
> When I adjusted the batch_size to 64,
> ![捕获](https://user-images.githubusercontent.com/30789945/90303604-e0475300-dee1-11ea-8a02-ae6debd810d6.JPG)
> It looks like "logits = linear_layer(nn.AvgPool2d(kernel_size=14)(conv_maps).view(-1, 512))" have some problem, and I don't understand what m1 and m2 mean here？

Hi, I think the error message from your screenshot is not related to the change of batch size. I am wondering if you have changed the networks from 'wideresnet18_places365.pth.tar' (as specified in line 28 of the file "dis_map_generation.py") to some others like resnet50. 

As shown from 'm2' in the error message, the expected input size for the linear_layer should be 2048 (but for wideresnet18_places365, this number should be 512, which is therefore matched with the '512' in 'm1').
So if you are changing the networks to others with expected input size of the linear_layer as 2048, you should accordingly change the  "logits = linear_layer(nn.AvgPool2d(kernel_size=14)(conv_maps).view(-1, 512))" to "logits = linear_layer(nn.AvgPool2d(kernel_size=14)(conv_maps).view(-1, 2048))" to make the sizes match.

**Comment by jizeyu** [comment]
I did use the resnet50 to train the model. According to your method, the problem has been solved. Thanks for your explain！

**Comment by ZhengyuZhao** [comment]
> I did use the resnet50 to train the model. According to your method, the problem has been solved. Thanks for your explain！

No problem! Thanks for your interests
